src/test/java/seedu/address/model/book/BookTest.java

```diff
@@ -1,22 +1,77 @@
+import static seedu.address.logic.commands.CommandTestUtil.VALID_BOOK_NAME_HARRY_POTTER;
 import static seedu.address.logic.commands.CommandTestUtil.VALID_BOOK_NAME_HUNGER_GAMES;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_ISBN_HARRY_POTTER;
 import static seedu.address.logic.commands.CommandTestUtil.VALID_ISBN_HARRY_POTTER_2;
 import static seedu.address.logic.commands.CommandTestUtil.VALID_ISBN_HUNGER_GAMES;
 import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_SCIFI;
+import static seedu.address.model.util.SampleDataUtil.SAMPLE_BOOK_CREATED_TIME;
 import static seedu.address.model.util.SampleDataUtil.getSampleBorrowedStatus;
 import static seedu.address.testutil.Assert.assertThrows;
 import static seedu.address.testutil.TypicalBooks.HARRY_POTTER;
 import static seedu.address.testutil.TypicalBooks.SEMAPHORE;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
 import org.junit.jupiter.api.Test;
 
+import seedu.address.model.tag.Tag;
 import seedu.address.testutil.BookBuilder;
 
 public class BookTest {
+    private static final BookName VALID_BOOK_NAME = new BookName(VALID_BOOK_NAME_HARRY_POTTER);
+    private static final BookStatus VALID_AVAILABLE_STATUS = BookStatus.createAvailableBookStatus();
+    private static final Isbn VALID_ISBN = new Isbn(VALID_ISBN_HARRY_POTTER);
+    private static final List<Author> VALID_AUTHORS = new ArrayList<>();
+    private static final Set<Tag> VALID_TAGS = new HashSet<>();
+    private static final BookStatus VALID_BORROWED_STATUS = getSampleBorrowedStatus();
+
+    @Test
+    public void constructor_nullBookName_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Book(null, VALID_ISBN, VALID_AUTHORS, VALID_TAGS,
+                SAMPLE_BOOK_CREATED_TIME, VALID_AVAILABLE_STATUS));
+    }
+
+    @Test
+    public void constructor_nullIsbn_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Book(VALID_BOOK_NAME, null, VALID_AUTHORS, VALID_TAGS,
+                SAMPLE_BOOK_CREATED_TIME, VALID_AVAILABLE_STATUS));
+    }
+
+    @Test
+    public void constructor_nullAuthors_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Book(VALID_BOOK_NAME, VALID_ISBN, null, VALID_TAGS,
+                SAMPLE_BOOK_CREATED_TIME, VALID_AVAILABLE_STATUS));
+    }
+
+    @Test
+    public void constructor_nullTags_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Book(VALID_BOOK_NAME, VALID_ISBN, VALID_AUTHORS, null,
+                SAMPLE_BOOK_CREATED_TIME, VALID_AVAILABLE_STATUS));
+    }
+
+    @Test
+    public void constructor_nullBookStatus_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Book(VALID_BOOK_NAME, VALID_ISBN, VALID_AUTHORS, VALID_TAGS,
+                SAMPLE_BOOK_CREATED_TIME, null));
+    }
```

Comment:
Good test cases to check if missing fields will throw error null pointer exception or not :100: